### PR TITLE
add TPU MNIST

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by kaushikb11** [comment]
@Borda What is required from my end here?

File: lightning_examples/mnist-tpu-training/.meta.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+title: TPU training with PyTorch Lightning
+author: PL team
+created: 2020-12-21
+updated: 2021-06-23
+license: CC BY-SA
+description: In this notebook, we'll train a model on TPUs. Changing one line of code is all you need to that.
```
Comment (kaushikb11) [inline lightning_examples/mnist-tpu-training/.meta.yml:6]:
```suggestion
description: In this notebook, we'll train a model on TPUs. Updating one Trainer flag is all you need for that.
```

**Comment by Borda** [comment]
> What is required from my end here?

as we do not have testing for TPU yet so mind verify that all are valued for collab, in particular, I feel a bit lost where is the TPU VM and where the new TPU Pod... so can we maybe make a section talking about how to install old VM and then new Pod?

File: lightning_examples/mnist-tpu-training/mnist-tpu.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+import torch.nn.functional as F
+from pytorch_lightning import LightningDataModule, Trainer
+from torch import nn
+from torch.utils.data import DataLoader, random_split
+from torchmetrics.functional import accuracy
+from torchvision import transforms
+# Note - you must have torchvision installed for this example
+from torchvision.datasets import MNIST
+
+BATCH_SIZE = 1024
```
Comment (kaushikb11) [inline lightning_examples/mnist-tpu-training/mnist-tpu.py:33]:
```suggestion
BATCH_SIZE = 32
```

**Review by kaushikb11 — APPROVED** [review]

File: lightning_examples/mnist-tpu-training/mnist-tpu.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+    def forward(self, x):
+        x = self.model(x)
+        return F.log_softmax(x, dim=1)
+
+    def training_step(self, batch, batch_idx):
+        x, y = batch
+        logits = self(x)
+        loss = F.nll_loss(logits, y)
+        self.log('train_loss', loss, prog_bar=False)
```
Comment (kaushikb11) [inline lightning_examples/mnist-tpu-training/mnist-tpu.py:110]:
```suggestion
        self.log('train_loss', loss)
```

File: lightning_examples/mnist-tpu-training/mnist-tpu.py (hunk 2 above)
Comment (Borda) [reply lightning_examples/mnist-tpu-training/mnist-tpu.py:33]: any reason for such a small?

Comment (kaushikb11) [reply lightning_examples/mnist-tpu-training/mnist-tpu.py:18]: Nvm, it's fine!

**Pull request merged**